File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-fixed-size/subarray-sum-fixed-size-k/code/cpp/subarray_sum_fixed_size_k_sol2.cpp
class Solution {
public:
    // Optimal: O(n) — Slide window, add/subtract at edges
    bool hasSubarraySum(vector<int>& nums, int k, int target) {
        int sum = 0;
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];                    // Add entering element
            if (i >= k) sum -= nums[i - k];    // Remove leaving element
            if (i >= k - 1 && sum == target) return true;
        }
        return false;
    }
};

File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-fixed-size/subarray-sum-fixed-size-k/code/python/subarray_sum_fixed_size_k_sol2.py
def has_subarray_sum(nums, k, target):
    window_sum = 0
    for i in range(len(nums)):
        window_sum += nums[i]
        if i >= k: window_sum -= nums[i - k]
        if i >= k - 1 and window_sum == target: return True
    return False